### ISSUE TITLE
Why SeqAn libraries are unsupported on macOS?

### ISSUE THREAD

**Issue by Hikoyu** [chanzuckerberg/shasta#95]
SeqAn libraries appears to be unused according to `AssemblerAlign1.cpp`.
```src/AssemblerAlign1.cpp
// Alternative alignment functions with 1 suffix.
#include "Assembler.hpp"
using namespace shasta;

// Standard library.
#include "chrono.hpp"

// Seqan.
#ifdef __linux__
#include <seqan/align.h>
#endif



#ifndef __linux__

// For macOS we don't have SeqAn, so we can't do any of this.
void Assembler::alignOrientedReads1(
    ReadId readId0, Strand strand0,
    ReadId readId1, Strand strand1,
    int matchScore,
    int mismatchScore,
    int gapScore)
{
    throw runtime_error("alignOrientedReads1 is not available on macOS.");
}

#else
```

But, SeqAn's libraries and header files are available via homebrew.
```
$ brew search seqan
==> Formulae
brewsci/bio/seqan
```
Are there any other problems for binary distribution?

**Comment by paoloczi** [comment]
The code in `Assembler::alignOrientedReads1` is experimental code that computes alignments of better quality, but is much slower. For that reason, it cannot be used in a real assembly. It can, however, be a useful debugging tool (e. g. you can use it when [exploring an assembly](https://chanzuckerberg.github.io/shasta/InspectingResults.html) using the Shasta http server).

Because this is not useful for production, I did not bother to activate it for macOS, as I do all of my testing and debugging on Linux. If you care to submit a Pull Request to activate it for macOS, I would be happy to merge it. If you do that, please make sure to add the required homebrew command to `shasta/scripts/InstallPrerequisites-macOS.sh`.

About a year ago I also used SeqAn for computing multiple sequence alignments for assembling marker graph edges, but I moved away from it because Spoa was much faster.

**Comment by Hikoyu** [comment]
@paoloczi I'm not sure when (how) to use Assembler::alignOrientedReads1. The compilation of shasta will be succeeded with SeqAn header files only. Are SeqAn libraries also needed?

**Comment by paoloczi** [comment]
As far as I know SeqAn does not need a library - header files are sufficient. If you look for example at  `shasta/staticExecutable/CMakeFile.txt` you will see that even for Linux there is no mention of SeqAn on the list of libraries to link with.

If you get a successful compilation and link, you are done. If you want to test, you can use command line options `--Align.alignMethodForReadGraph 1` and/or `--Align.alignMethodForMarkerGraph 1`. Selecting 0 (the default) uses the Shasta rough alignment algorithm, and selecting 1 uses SeqAn (`Assembler::alignOrientedReads1`). 

You can also use the Shasta http server (see [here](https://chanzuckerberg.github.io/shasta/InspectingResults.html) for more information) to trigger use of `Assembler::alignOrientedReads1`. From the browser, click on `Alignments`, then `Align two reads`, and select `Use method 1`.

**Comment by Nheyer** [comment]
I found that SeqAn @ 2 is needed for shasta to compile, and the most updated SeqAn@3 will not work

**Comment by paoloczi** [comment]
@Nheyer for that reason, `InstallPrerequisites-Ubuntu.sh` installs SeqAn 2.4.0. Is that a problem?

**Comment by Nheyer** [comment]
I don't see InstallPrerequisites-Ubuntu.sh

**Comment by paoloczi** [comment]
No, per the [documentation](https://chanzuckerberg.github.io/shasta/BuildingFromSource.html), building Shasta is only supported on Ubuntu 16.04, Ubuntu 18.04, and macOS.

You can run Shasta on CentOS, but not build it there. You can easily download either a [release](https://github.com/chanzuckerberg/shasta/releases) or a current test build (see [here](https://chanzuckerberg.github.io/shasta/BuildingFromSource.html#DownloadTestBuild) for more information). The `shasta-Linux` executable is statically linked and works on most current Linux versions, including CentOS, without requiring installation (just do `chmod` after download to make it executable). If you are on an old Linux kernel, the `shasta-Linux` executable will not work. In that case, download the `shasta-OldLinux` executable instead.

**Comment by paoloczi** [comment]
This was fixed by commit faad63cb6486db22cabd5ab2a788eff9648292dd. SeqAn functionality, is not fully included in the macOS builds.
